tests/test_monkeypatch.py

```diff
@@ -1,17 +1,64 @@
+from django.contrib.sites.models import Site
+
+from cms import api
 from cms.cms_toolbars import LANGUAGE_MENU_IDENTIFIER
+from cms.extensions.extension_pool import ExtensionPool
 from cms.test_utils.testcases import CMSTestCase
 from cms.toolbar.toolbar import CMSToolbar
 from cms.toolbar.utils import get_object_edit_url
 from cms.utils.urlutils import admin_reverse
 
 from djangocms_versioning.plugin_rendering import VersionContentRenderer
+from djangocms_versioning.test_utils.extensions.models import (
+    TestPageExtension,
+    TestTitleExtension,
+)
 from djangocms_versioning.test_utils.factories import (
     PageContentFactory,
     PageVersionFactory,
     PollVersionFactory,
+    UserFactory,
 )
 
 
+class MonkeypatchExtensionTestCase(CMSTestCase):
+    def setUp(self):
+        user = UserFactory()
+        self.version = PageVersionFactory(content__language="en")
+        pagecontent = PageContentFactory(
+            page=self.version.content.page, language="de"
+        )
+        self.page = self.version.content.page
+        site = Site.objects.first()
+        self.new_page = self.page.copy(
+            site=site,
+            parent_node=self.page.node.parent,
+            translations=False,
+            permissions=False,
+            extensions=False,
+        )
+
+        new_page_content = api.create_title(
```

Comment:
Strange to use factories and then the api for creating a new PageContent item. Why is this?

Reply:
Did not work properly with factories, and using api.create_title is in itself a kind of factory, no? :)

Reply:
Create title creates the models themselves unlike a factory which only creates what you specify for it to create.

Reply:
I think as you are testing the actual functionality of what happens with the slug and path on creation and changes it is ok to use the actual models vs the factories. 

We do have a PageUrl factory for tests that need to access a page.